extensions/wikia/ArticleVideo/scripts/recommended-video.jwplayer.js

```diff
@@ -0,0 +1,206 @@
+require([
+	'jquery',
+	'wikia.tracker',
+	'wikia.articleVideo.featuredVideo.cookies',
+	'wikia.loader',
+	'wikia.mustache'
+], function ($, tracker, featuredVideoCookieService, loader, mustache) {
+	'use strict';
+
+	var recommendedVideoSelector = '.article-recommended-video .video',
+		jwPlaylistDataUrl = 'https://cdn.jwplayer.com/v2/playlists/',
+		windowHeight = Math.max(document.documentElement.clientHeight, window.innerHeight || 0),
+		$unit = $('.article-recommended-video-unit'),
+		currentVideo = null,
+		recommendedVideoElementId = 'recommendedVideo',
+		recommendedVideoData = null,
+		player = null,
+		$actualVideo = null,
+		currentItemNumber = 1,
+		isExpanded = false;
+
+	function reveal() {
+		var scrollTop = $(window).scrollTop();
+
+		if (scrollTop > windowHeight * 1.5) {
+			$unit.addClass('revealed');
+			window.wikiaJWPlayer(
+				recommendedVideoElementId,
+				getPlayerSetup(recommendedVideoData),
+				onPlayerReady
+			);
+
+			tracker.track({
+				category: 'recommended-video',
+				trackingMethod: 'both',
+				action: tracker.ACTIONS.VIEW,
+				label: 'recommended-video-revealed'
+			});
+		}
+	}
+
+	function loadAssets() {
+		loader({
+			type: loader.MULTI,
+			resources: {
+				mustache: 'extensions/wikia/ArticleVideo/templates/ArticleVideo_recommendedVideos.mustache'
+			}
+		}).done(function (assets) {
+			renderView(assets.mustache[0]);
+			window.setTimeout(reveal, 5000);
+		});
+	}
+
+	function renderView(view) {
+		// This is needed in order to access item index in mustache
+		recommendedVideoData.playlist.forEach(function (video, index) {
+			video.index = index;
+
+			return video;
+		});
+		$unit.find('.article-recommended-videos').html(
+			mustache.render(view, {
+				playlist: recommendedVideoData.playlist,
+				firstVideo: recommendedVideoData.playlist[0]
+			})
+		);
+
+		$actualVideo = $unit.find('.actual-video');
+	}
+
+	function onJWDataLoaded(videoElementId, jwPlayerData) {
+		recommendedVideoData = jwPlayerData;
+		recommendedVideoElementId = videoElementId;
+
+		loadAssets();
+	}
+
+	function onCloseClicked() {
+		$unit.removeClass('revealed');
+	}
+
+	function playItem(data) {
+		currentItemNumber = data.index + 1;
+
+		$unit
+			.removeClass('plays-video-1 plays-video-2 plays-video-3 plays-video-4 plays-video-5')
+			.addClass('plays-video-' + currentItemNumber);
+
+		$actualVideo.find('h3').html(data.item.title);
+
+		tracker.track({
+			category: 'recommended-video',
+			trackingMethod: 'both',
+			action: tracker.ACTIONS.VIEW,
+			label: 'playlist-item-start'
+		});
+	}
+
+	function init() {
+		if ($unit && window.wikiaJWPlayer) {
```

Comment:
since `$unit` is a jquery element, it'll be a jQuery wrapper even if nothing will be found - consider `$unit.length`